cpp/src/fil/infer.cu

```diff
@@ -350,7 +350,7 @@ struct tree_aggregator_t<NITEMS, CATEGORICAL_LEAF> {
                                                             int num_rows) {
     __syncthreads();
     int item = threadIdx.x;
-    int row = blockIdx.x * NITEMS + item;
+    int row = item;
```

Comment:
No need to introduce a new variable, just `row` (instead of `item`) would be enough.